src/controllers/unreal.rs

```diff
@@ -70,96 +68,171 @@ impl UnrealCameraBundle {
+    let mut controller = if let Some(controller) = controllers.iter_mut().next() {
         controller
     } else {
         return;
     };
     let UnrealCameraController {
         enabled,
+        rotate_sensitivity: mouse_rotate_sensitivity,
         mouse_translate_sensitivity,
-        mouse_rotate_sensitivity,
-        trackpad_translate_sensitivity,
+        wheel_translate_sensitivity,
+        mut keyboard_mvmt_sensitivity,
+        keyboard_mvmt_wheel_sensitivity,
         ..
     } = *controller;
 
     if !enabled {
         return;
     }
 
+    let left_pressed = mouse_buttons.pressed(MouseButton::Left);
+    let right_pressed = mouse_buttons.pressed(MouseButton::Right);
+    let middle_pressed = mouse_buttons.pressed(MouseButton::Middle);
+
     let mut cursor_delta = Vec2::ZERO;
     for event in mouse_motion_events.iter() {
         cursor_delta += event.delta;
     }
 
-    match (
-        mouse_buttons.pressed(MouseButton::Left),
-        mouse_buttons.pressed(MouseButton::Right),
-    ) {
-        (true, true) => {
-            events.send(ControlEvent::TranslateEye(
-                mouse_translate_sensitivity * cursor_delta,
-            ));
-        }
-        (true, false) => {
-            events.send(ControlEvent::Locomotion(Vec2::new(
-                mouse_rotate_sensitivity.x * cursor_delta.x,
-                mouse_translate_sensitivity.y * cursor_delta.y,
-            )));
+    let mut wheel_delta = 0.0;
+    for event in mouse_wheel_reader.iter() {
+        wheel_delta += event.x + event.y;
+    }
+
+    let mut panning_dir = Vec2::ZERO;
+    let mut translation_dir = Vec2::ZERO; // y is forward/backward axis, x is rotation around Z
+
+    for key in keyboard.get_pressed() {
+        match key {
+            KeyCode::E => {
+                panning_dir.y += 1.0;
+            }
+
+            KeyCode::Q => {
+                panning_dir.y -= 1.0;
+            }
+
+            KeyCode::A => {
+                panning_dir.x -= 1.0;
+            }
+
+            KeyCode::D => {
+                panning_dir.x += 1.0;
+            }
+
+            KeyCode::S => {
+                translation_dir.y -= 1.0;
+            }
+
+            KeyCode::W => {
+                translation_dir.y += 1.0;
+            }
+
+            _ => {}
         }
-        (false, true) => {
-            events.send(ControlEvent::Rotate(
-                mouse_rotate_sensitivity * cursor_delta,
-            ));
+    }
+
+    let mut panning = Vec2::ZERO;
+    let mut locomotion = Vec2::ZERO;
+
+    // If any of the mouse button are pressed; read additional signals from the keyboard for panning
+    // and locomotion along camera view axis
+    if left_pressed || middle_pressed || right_pressed {
+        panning += keyboard_mvmt_sensitivity * panning_dir;
+
+        if translation_dir.y != 0.0 {
+            locomotion.y += keyboard_mvmt_sensitivity * translation_dir.y;
         }
```

Comment:
Seems like this if statement is not necessary.

Reply:
Indeed, no need for the condition, because we'd just multiply by zero. Probably a remnant of previous iterations I did where I had other statements in that if!